examples/seldon_custom_pytorch_deployment/pytorch_seldon_deploy.py

```diff
@@ -0,0 +1,284 @@
+
+        print("Train Epoch: {} \tLoss: {:.6f}".format(epoch, loss.item()))
+
+    return model
+
+
+class EvaluatorConfig(BaseStepConfig):
+    """Trainer params"""
+
+    batch_size: int = 4
+    shuffle: bool = True
+
+
+@step
+def torch_evaluator(
+    config: EvaluatorConfig,
+    model: nn.Module,
+) -> float:
+    """Calculate the loss for the model for each epoch in a graph"""
+    model.eval()
+    test_loader = data_loader(
+        train_set=False, batch_size=config.batch_size, shuffle=config.shuffle
+    )
+    test_loss = 0
+    correct = 0
+    with torch.no_grad():
+        for data, target in test_loader:
+            data, target = data.to(DEVICE), target.to(DEVICE)
+            output = model(data)
+            test_loss += F.nll_loss(
+                output, target, reduction="sum"
+            ).item()  # sum up batch loss
+            pred = output.argmax(
+                dim=1, keepdim=True
+            )  # get the index of the max log-probability
+            correct += pred.eq(target.view_as(pred)).sum().item()
+
+    test_loss /= len(test_loader.dataset)
+
+    print(
+        "\nTest set: Average loss: {:.4f}, Accuracy: {}/{} ({:.0f}%)\n".format(
+            test_loss,
+            correct,
+            len(test_loader.dataset),
+            100.0 * correct / len(test_loader.dataset),
+        )
+    )
+    return correct / len(test_loader.dataset)
+
+
+class DeploymentTriggerConfig(BaseStepConfig):
+    """Parameters that are used to trigger the deployment"""
+
+    min_accuracy: float
+
+
+@step
+def deployment_trigger(
+    accuracy: float,
+    config: DeploymentTriggerConfig,
+) -> bool:
+    """Implements a simple model deployment trigger that looks at the
+    input model accuracy and decides if it is good enough to deploy"""
+
+    return accuracy > config.min_accuracy
+
+
+# Define the pipeline
+@pipeline(
+    enable_cache=True,
+    required_integrations=[SELDON, PYTORCH],
+    requirements_file=requirements_file,
+)
+def seldon_pytorch_deployment_pipeline(
+    trainer,
+    evaluator,
+    deployment_trigger,
+    custom_model_deployer,
+):
+    # Link all the steps artifacts together
+    model = trainer()
+    accuracy = evaluator(model=model)
+    deployment_decision = deployment_trigger(accuracy=accuracy)
+    custom_model_deployer(
+        deployment_decision,
+        model,
+    )
+
+
+class MnistPytorch(ZenMLCustomModel):
+    """Custom model deployment class
+
+    This class is used to define the custom model deployment logic.
+    which describe how to load the model, how to run predictions and
+    maybe extra logic to handle pre-processing or post-processing.
+    """
+
+    def __init__(self, model_uri: str = None, *args: Any, **kwargs: Any):
+        super().__init__(model_uri=model_uri)
+
+    def load(self):
+        """Loads the model into memory.
+
+        This method is called before the first prediction.
+        and it requires understanding of materializer and how
```

Comment:
```suggestion
        and it requires understanding the materializer and how
```

Also, I'm not fully sure what this means...